### DISCUSSION TITLE
Vous ne devriez pas faire tourner votre serveur de mail parce que c'est dur · poolp.org

### DISCUSSION THREAD

**Discussion by giscus** [poolpOrg/poolp.org#166] [Articles]
# Vous ne devriez pas faire tourner votre serveur de mail parce que c'est dur · poolp.org

TL;DR: Le mail n’est pas dur: les gens répètent ça parce qu’ils l’ont lu, pas parce qu’ils ont essayé Les Big Mailer Corps sont content de ce mythe, ça fait grossir leur base d’utilisateurs Les Big Mailer Corps contrôlent un large pourcentage de l’espace d’adressage des e-mails ce qui n’est bon pour personne C’est ok que des gens aient leurs adresses e-mails hébergés chez des Big Mailer Corps tant qu’il y a assez de gens en dehors EDIT (2019-12-15) # Un guide pratique pour installer un serveur de mail a été publié sur ce blog.

https://poolp.org/fr/posts/2019-08-30/vous-ne-devriez-pas-faire-tourner-votre-serveur-de-mail-parce-que-cest-dur/

**Comment by l4nj1n9** [comment]
Je suis content de lire ça. Pour ma part, j'avoue ne pas m'être encore résolu à me lancer dans l'aventure, mais je suis résolu à le faire dans les mois à venir. Il est vrai que j'ai été longtemps découragé par la difficulté de la chose, mais ce n'était pas le principal obstacle. Pour moi, le gros problème est l'importance critique du service. Je veux bien auto-héberger mes mails, mais que faire en cas de panne si je suis dans l'impossibilité d'intervenir rapidement ? Cet argument n'est pas évoqué dans l'article.

À deux reprises, mon point de vue a évolué. D'abord, j'ai eu recours aux services de mon registrar pour héberger mes mails plutôt qu'un prestataire spécialisé et gratuit. Du coup, j'ai découvert SPF, DKIM et je me suis dit que les choses n'étaient peut-être pas si compliquées. Puis mon registrar a décidé de changer de politique pour adopter des tarifs prohibitifs à mes yeux. Je me suis alors dit, bon, lançons-nous !

Cet article m'y encourage et je t'en remercie ! Hâte d'en apprendre plus !

**Reply by ouafnico** [reply]
Le côté critique du service en effet est une bonne interrogation. Mais il y a pas mal de solutions.

La plus simple est d'avoir plusieurs SMTP, avec plusieurs priorités dans les DNS "MX". 
Chaque SMTP doit gérer le domaine mail, et devra transmettre le mail au serveur MDA (dovecot par exemple) dès qu'il est joignable (via LMTP par exemple, over vpn s'il faut, ..).
Le but est de garder les mails reçus et ne pas les perdre, ça permet de tamponner sur l'urgence quand notre auto-hébergement est indispo.

Le MDA, comme dovecot, sait très bien gérer en 2 lignes de conf, un cluster de deux noeuds, qui permet aussi de gérer ce soucis :) 

J'en donne un exemple ici : https://ouafnico.shivaserv.fr/posts/infra-mail/

**Comment by ouafnico** [comment]
Ah ah ça fait plaisir.
J'héberge mes mails depuis quelque chose comme 15 ans ou plus. J'en avais même fait un article https://ouafnico.shivaserv.fr/posts/infra-mail/

C'est difficile d'avoir une conf entièrement parfaite, surtout au niveau des spams, mais en effet, une fois qu'on a du SPF, du DKIM et un rDNS valable, je n'ai jamais eu un souci.

Un bon truc pour valider sa conf c'est mail-tester.com.

Le spam, j'en suis devenu rigide : via fail2ban, le moindre accès sur mon SMTP qui est mal parti, l'ip publique est bannie pendant des semaines :D

En tout cas j'aime bien la façon dont t'aborde le sujet, car même dans mon métier d'admin, j'entends beaucoup du monde dire que les mails c'est trop compliqué.

**Comment by guldnerthomas** [comment]
Merci pour cette analyse intéressante et pratiquement a contre tendance. Je fais partie de ces gens qui suivent l'actualité IT et qui tombent sur des articles de blog souvent au travers du https:/www.journalduhackernet/ et je reconnais que sur cette dernière année je suis tombé (au moins deux fois de mémoire) sur des articles indiquant que faire du mail auto hébergé, c'est l'enfer. Du coup ça ne m'a jamais encouragé a le faire et ça m'arrangeait jusqu'à présent puisque j'ai toujours une tonne d'autre chose a faire...
Ton article va me faire revoir ma réflexion et peut être qu'un jour j'aurais le mien.
En tout cas je ne colporterais plus que le mail c'est compliqué sans avoir testé de manière approfondie.

**Reply by poolpOrg** [reply]
Pour chaque personne qui répète "le mail c'est dur", sans forcément avoir essayé, je trouve une personne qui fait réellement tourner un serveur et peut témoigner que sa maintenance est minimale et qu'il rencontre moins d'un problème par an.

En tant qu'ancien développeur sur OpenSMTPD, j'ai reçu une tonne de mails au fil des années de personnes qui m'ont écrit spontanément pour me dire que c'est un plaisir de gérer leur serveur, c'est vraiment aux antipodes de l'enfer qu'on décrit.

OUI, il y a du travail et un coût d'entrée pour la mise en place et la construction d'une réputation, ce qui veut dire que pendant une période ça peut passer un peu moins bien chez l'un ou l'autre des hébergeurs le temps d'avoir un rythme de croisière et des échanges réguliers, mais entre ça et abandonner ses communications à une boite privée qui se fait plaisir dessus, y a pas photo.

Merci pour le lien vers journalduhacker.net, je ne connaissais pas <3

**Comment by masao77** [comment]
J'approuve cet article, je vais tout de même nuancer un peu.

Tu parles de SMTP, d'antispam et de SPF/DKIM/rDNS. Pour une solution complète il faut rajouter un webmail, des filtres (sieve/managesieve) et éventuellement du CalDAV/CardDAV. Et il faut pouvoir backuper/restaurer tout ça. Rien d'insurmontable pour tout sysadmin qui se respecte, mais pour un débutant, ça peut faire pas mal de choses à appréhender (même si une fois mis en place, il n'y a que peu de maintenance à faire).

Comme dit dans un autre commentaire, il faut aussi prendre en compte la criticité. Selon les besoins un simple MX secondaire peut suffire. S'il faut redonder toute la solution, c'est plus complexe (même si pas infaisable). C'est la raison pour laquelle je n'héberge que mon mail à moi, ainsi ça ne dérange personne si le serveur est KO pendant que je suis parti en vacances.

Pour ceux qui ne se sentent pas tout de suite de se lancer dans l'aventure, vous pouvez simplement prendre un nom de domaine en faisant toujours héberger votre mail par un provider. Ainsi, si un jour vous souhaitez changer de provider parce qu'il ne vous plait plus (ou ferme), ou que vous souhaitez vous autohéberger, vous pourrez plus facilement migrer.
